mainwindow.cpp

```diff
@@ -577,78 +513,96 @@ QStringList MainWindow::getAttackParameters()
     } else {
         // In case we do not have explicit format we erase remembered
         // key.
-        m_format = "";
+        m_format.clear();
     }
+    m_settings.setValue("formatJohn", m_format);
+    m_settings.setValue("formatUI", m_ui->comboBox_Format->currentText());
+    
     // Modes
     QWidget* selectedMode = m_ui->attackModeTabWidget->currentWidget();
     if (selectedMode == m_ui->defaultModeTab) {
         // Default behaviour - no modes
         // There are no options here.
+        m_settings.setValue("mode", "default");
     } else if (selectedMode == m_ui->singleModeTab) {
         // "Single crack" mode
         parameters << "--single";
+        m_settings.setValue("mode", "single");
         // External mode, filter
-        if (m_ui->checkBox_SingleCrackModeExternalName->isChecked())
+        if (m_ui->checkBox_SingleCrackModeExternalName->isChecked()) {
```

Comment:
I think I mentioned in an e-mail discussing that we should follow the Qt coding style with the only possible exception being opening curly braces on a new line, or at least that I prefer it. So let's follow that form now on. No need to change everything now, just as you go.
I updated the openwall wiki page with that and also added this Github wiki page on the matter:
https://github.com/shinnok/johnny/wiki/Development